packages/typescript-estree/tests/lib/__snapshots__/typescript.ts.snap

```diff
@@ -64986,619 +65830,5232 @@ Object {
+          "line": 2,
         },
       },
       "range": Array [
-        38,
-        39,
+        74,
+        75,
       ],
       "type": "Punctuator",
-      "value": "{",
+      "value": ";",
     },
     Object {
       "loc": Object {
         "end": Object {
-          "column": 4,
-          "line": 5,
+          "column": 5,
+          "line": 3,
         },
         "start": Object {
-          "column": 3,
-          "line": 5,
+          "column": 2,
+          "line": 3,
         },
       },
       "range": Array [
-        40,
-        41,
+        78,
+        81,
       ],
-      "type": "Punctuator",
-      "value": "[",
+      "type": "Keyword",
+      "value": "let",
     },
     Object {
       "loc": Object {
         "end": Object {
-          "column": 8,
-          "line": 5,
+          "column": 10,
+          "line": 3,
         },
         "start": Object {
-          "column": 4,
-          "line": 5,
+          "column": 6,
+          "line": 3,
         },
       },
       "range": Array [
-        41,
-        45,
+        82,
+        86,
       ],
-      "type": "String",
-      "value": "'__'",
+      "type": "Identifier",
+      "value": "find",
     },
     Object {
       "loc": Object {
         "end": Object {
-          "column": 9,
-          "line": 5,
+          "column": 12,
+          "line": 3,
         },
         "start": Object {
-          "column": 8,
-          "line": 5,
+          "column": 11,
+          "line": 3,
         },
       },
       "range": Array [
-        45,
-        46,
+        87,
+        88,
       ],
       "type": "Punctuator",
-      "value": "]",
+      "value": "=",
     },
     Object {
       "loc": Object {
         "end": Object {
-          "column": 10,
-          "line": 5,
+          "column": 14,
+          "line": 3,
         },
         "start": Object {
-          "column": 9,
-          "line": 5,
+          "column": 13,
+          "line": 3,
         },
       },
       "range": Array [
-        46,
-        47,
+        89,
+        90,
       ],
       "type": "Punctuator",
-      "value": ":",
+      "value": "(",
     },
     Object {
       "loc": Object {
         "end": Object {
           "column": 15,
-          "line": 5,
+          "line": 3,
         },
         "start": Object {
-          "column": 11,
-          "line": 5,
+          "column": 14,
+          "line": 3,
         },
       },
       "range": Array [
-        48,
-        52,
+        90,
+        91,
       ],
-      "type": "Keyword",
-      "value": "null",
+      "type": "Identifier",
+      "value": "s",
     },
     Object {
       "loc": Object {
         "end": Object {
-          "column": 17,
+          "column": 18,
+          "line": 3,
+        },
+        "start": Object {
+          "column": 16,
+          "line": 3,
+        },
+      },
+      "range": Array [
+        92,
+        94,
+      ],
+      "type": "Punctuator",
+      "value": "??",
```

Comment:
This should be a `LogicalExpression` not a `Punctuator` (unless this is testing the lexer instead of the tokenizer?)

Reply:
That particular chunk of the snapshot is the tokens list that gets emitted as part of the root AST.

Reply:
They are `LogicalExpression`s in `babel-eslint`
![image](https://user-images.githubusercontent.com/73085/66366336-0bf8d300-e9cb-11e9-8c1c-7a27d3755e53.png)

In the spec language, it is defined under the "Binary Logical Operators" heading https://tc39.es/proposal-nullish-coalescing/ but as a different kind of expression (`CoalesceExpression`).

(if you were confused, like I was, `BitwiseORExpression` is just the "bottom type" (lowest precedence) expression that is allowed in that position, see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Operator_Precedence; this is how, syntactically, it requires that combining `&&` or `||` with `??` be disambiguated with parens, as there are no productions involving `??` that allow a `LogicalANDExpression` or `LogicalORExpression` to fit other than through `CoverParenthesizedExpressionAndArrowParameterList` (i.e. the use of parens))